storage-node/src/storage_reader/s3.rs

```diff
@@ -102,27 +109,33 @@ impl Stream for S3DataStream {
             match ready!(data_fut.poll_unpin(cx)) {
                 Ok(bytes) => {
                     self.data_fut.take();
-                    self.buffer.extend(bytes.into_bytes());
+                    // TODO: extend is real `copy`, expensive! Can we directly turn AggregateBytes to BytesMut?
+                    let data = bytes.into_bytes();
```

Comment:
We have to copy the returned bytes anyway?

Reply:
I am not sure, since I think if the bytes are already in the memory, then maybe we can make buffer_inner (BytesMut) point to it if there is no remaining data in the buffer_inner.

Reply:
Actually currently I don’t think chunksize is useful, after all the data is already in the memory (chunksize seems to add more cost of disk I/O and poll next). But extend is truly expensive, so we have to copy all the data in the current code, I think we should avoid this. If chunksize is useful, I would suggest to find some workarounds to solve this.

Reply:
+1. If it's not useful as we move forward with the project we can remove it.